### DISCUSSION TITLE
variable number of columns

### DISCUSSION THREAD

**Discussion by StefanRiess** [grid-js/gridjs#1280] [Q&A]
All examples I have found so far use a fixed (known in advance) list of columns. 
const grid = new Grid({
  columns: ['Name', 'Email', 'Phone Number'], .....
Is there a way to supply the list of columns when rendering the html? To stay within this example I have sometimes a phone-Numer at home or at work or several cell-phone numbers.

Does anyone have a reference for me? Thanks in advance

**Comment by rendorHaevyn** [comment]
100% - the `data` element is dynamic if a `server` call is made, and subsequently, we need to be able to dynamically also populate the `columns` element, based on that api call (ie: results.data, results.columns, etc)

**Comment by leehrat** [comment]
Yeah I haven't found a way to do this either.
